.tekton/scripts/check-partner-tasks.sh

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env bash
+
+SCRIPTDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+cd "$SCRIPTDIR/../.." || exit 1
+
+# Expected partner task directory structure:
+# task/partners/<task name>/<version>/<task file>
+
+if [ ! -e "partners/" ]; then
+  exit 0
+fi
+
+check_dir_structure() {
+  issue_found=0
+  for task_dir in partners/*; do
+    task_name=$(basename "$task_dir")
+    for version_dir in "$task_dir"/*; do
+      task_version=$(basename "$version_dir")
+      if ! echo "$task_version" | grep "^[0-9.]\+$" >/dev/null 2>&1; then
+        echo "error: $version_dir is not a version directory. A version directory name only contains digits and dot, e.g. 0.1, 0.2.1"
```

Comment:
I am wondering .. Do we expect the partners to look into the PR check failure logs to figure out what failed or we can communicate the the error messages to the partner submitter through the comments in the PR.

Reply:
Good question. What you are thinking is an ideal solution for partner to check out the potential failure messages. That would be good to be implemented in a separate JIRA story, and it is not only for the partner task submission, but it can also be an general solution for showing pipeline summary information to users.

Reply:
Related to my question, I see this statement `CI tests do the due diligence on the changes proposed in the PR. Success/Failures are reported in a way that the PR author can take reasonable action to resolve the issue.` in the ADR.

Reply:
Since we created a improvement story to address this. Closing this for this PR.